modules/registry/registry.go

```diff
@@ -24,17 +35,56 @@ type RegistryContainer struct {
 
 // Address returns the address of the Registry container, using the HTTP protocol
 func (c *RegistryContainer) Address(ctx context.Context) (string, error) {
-	port, err := c.MappedPort(ctx, "5000")
+	host, err := c.HostAddress(ctx)
 	if err != nil {
 		return "", err
 	}
 
-	ipAddress, err := c.Host(ctx)
+	return "http://" + host, nil
+}
+
+// HostAddress returns the host address including port of the Registry container.
+func (c *RegistryContainer) HostAddress(ctx context.Context) (string, error) {
+	port, err := c.MappedPort(ctx, registryPort)
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("mapped port: %w", err)
+	}
+
+	host, err := c.Container.Host(ctx)
+	if err != nil {
+		return "", fmt.Errorf("host: %w", err)
+	}
+
+	if host == "localhost" {
+		// This is a workaround for WSL, where localhost is not reachable from Docker.
+		host, err = localAddress(ctx)
+		if err != nil {
+			return "", fmt.Errorf("local ip: %w", err)
+		}
 	}
 
-	return fmt.Sprintf("http://%s:%s", ipAddress, port.Port()), nil
+	return net.JoinHostPort(host, port.Port()), nil
+}
+
+// localAddress returns the local address of the machine
+// which can be used to connect to the local registry.
+// This avoids the issues with localhost on WSL.
+func localAddress(ctx context.Context) (string, error) {
```

Comment:
Do you think that https://github.com/testcontainers/testcontainers-go/pull/2403 has something to do here? It will apply to all containers

Reply:
This is really just a bug in WSL, there's no reason it shouldn't work it just doesn't.